src/agent/coverage/src/code.rs

```diff
@@ -302,83 +301,113 @@ impl Region for Symbol {
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(transparent)]
-pub struct SymbolFilterSpec {
-    filters: HashMap<String, Filter>,
+pub struct CmdFilterDef {
+    defs: Vec<ModuleRuleDef>,
 }
 
-#[derive(Clone, Debug)]
-pub struct SymbolFilter {
-    /// Pre-compiled regex set for fast matching.
-    regexes: RegexSet,
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct ModuleRuleDef {
+    pub module: String,
 
-    /// Maps module names (via regexes) to filters for the symbol names provided
-    /// by the matched module(s).
+    #[serde(flatten)]
+    pub rule: RuleDef,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+enum RuleDef {
+    Include { include: bool },
+    Exclude { exclude: bool },
+    Filter(Box<Filter>),
+}
+
+#[derive(Clone, Debug)]
+enum Rule {
+    /// Asserts that the entire sould be be tracked (and its symbols included),
+    /// or ignored, and its symbols excluded.
     ///
-    /// The integer key to this map is an index into the `regexes` field, and so
-    /// identifies a (module-matching) regex in that set.
-    filters: BTreeMap<usize, Filter>,
+    /// The implied symbol tracking behavior could be encoded by a filter, but a
+    /// distinction at this level lets us avoid parsing modules that we want to
```

Comment:
@bmc-msft, see this comment for one motivation behind allowing `"exclude": true`. I also find that to be more direct (from a user perspective) than asking them to encode that behavior with the right regex-based rule.

We could say "use regex lists, only", at the expense of some static information. The cost there isn't super high. Worst case it means we'd attempt to parse every module that is linked or loaded by the target. But we only do that once per task (due to our cache), so it is definitely amortized.